### PR TITLE
16256: Properly get tab origin for brave shields.

### PR DESCRIPTION
**Pull request by iefremov**
*MANUAL* Uplift of #9038
Fixes https://github.com/brave/brave-browser/issues/16265

Pre-approval checklist: 
- [x] You have tested your change on Nightly. 
- [ ] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by iefremov** [comment]
@mariospr could you please take an extra look? This is a manual uplift

**Comment by kjozwiak** [comment]
Restarting `Linux` due to timing out as per https://ci.brave.com/job/pr-brave-browser-manual_pr9038_ie_tab_origin_1.25.x-linux/1/console.

```
11:19:05  [187/227] STAMP obj/brave/brave_dist_resources.stamp
Body did not finish within grace period; terminating with extreme prejudice
```

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.25.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/9038#issuecomment-858333274.

**Pull request merged**
